File: src/main/java/com/temelyan/pomoapp/util/ExceptionHandler/ErrorInfo.java
package com.temelyan.pomoapp.util.ExceptionHandler;

import java.util.Arrays;

public class ErrorInfo {
    private final String url;
    private final String type;
    private final String typeMessage;
    private final String[] details;

    public ErrorInfo(CharSequence url, String type, String typeMessage, String... details) {
        this.url = url.toString();
        this.type = type;
        this.typeMessage = typeMessage;
        this.details = details;
    }

    @Override
    public String toString() {
        return "ErrorInfo{" +
                "url='" + url + '\'' +
                ", type='" + type + '\'' +
                ", typeMessage='" + typeMessage + '\'' +
                ", details=" + Arrays.toString(details) +
                '}';
    }
}

File: src/main/java/com/temelyan/pomoapp/util/ExceptionHandler/ErrorInfo.kt
package com.temelyan.pomoapp.util.ExceptionHandler

import java.util.*

class ErrorInfo(url: CharSequence, private val type: String, private val typeMessage: String, vararg details: String) {
    private val url: String
    private val details: Array<out String>

    init {
        this.url = url.toString()
        this.details = details
    }

    override fun toString(): String {
        return "ErrorInfo{" +
                "url='" + url + '\''.toString() +
                ", type='" + type + '\''.toString() +
                ", typeMessage='" + typeMessage + '\''.toString() +
                ", details=" + Arrays.toString(details) +
                '}'.toString()
    }
}
